src/components/ErrorBoundary/__tests__/prepareErrorStack.test.ts

```diff
@@ -0,0 +1,76 @@
+    at workLoopSync (http://localhost:3000/static/js/bundle.js:108422:9)
+    at renderRootSync (http://localhost:3000/static/js/bundle.js:108395:11)
+    at recoverFromConcurrentError (http://localhost:3000/static/js/bundle.js:107887:24)
+    at performConcurrentWorkOnRoot (http://localhost:3000/static/js/bundle.js:107800:26)`;
+
+        const preparedStack = `TypeError: Cannot read properties of null (reading 'hello')
+    at Content (/static/js/bundle.js:4728:8)
+    at renderWithHooks (/static/js/bundle.js:98996:22)
+    at mountIndeterminateComponent (/static/js/bundle.js:102967:17)
+    at beginWork (/static/js/bundle.js:104270:20)
+    at beginWork$1 (/static/js/bundle.js:109229:18)
+    at performUnitOfWork (/static/js/bundle.js:108499:16)
+    at workLoopSync (/static/js/bundle.js:108422:9)
+    at renderRootSync (/static/js/bundle.js:108395:11)
+    at recoverFromConcurrentError (/static/js/bundle.js:107887:24)
+    at performConcurrentWorkOnRoot (/static/js/bundle.js:107800:26)`;
+
+        // Mock window.location.origin that is used inside prepareErrorStack
+        const windowSpy = jest.spyOn(window, 'window', 'get');
+        windowSpy.mockImplementation(() => {
+            return {
+                location: {
+                    origin: 'http://localhost:3000',
+                },
+            } as Window & typeof globalThis;
+        });
+
+        expect(prepareErrorStack(stack, {trim: false, maxLength: undefined})).toBe(preparedStack);
+
+        windowSpy.mockRestore();
+    });
+
+    test('Limit trace to maxLenght', () => {
+        const stack = `TypeError: Cannot read properties of null (reading 'hello')
+    at Content (/static/js/bundle.js:4725:8)
+    at renderWithHooks (/static/js/bundle.js:98993:22)
+    at mountIndeterminateComponent (/static/js/bundle.js:102964:17)
+    at beginWork (/static/js/bundle.js:104267:20)
+    at beginWork$1 (/static/js/bundle.js:109226:18)
+    at performUnitOfWork (/static/js/bundle.js:108496:16)
+    at workLoopSync (/static/js/bundle.js:108419:9)
+    at renderRootSync (/static/js/bundle.js:108392:11)
+    at recoverFromConcurrentError (/static/js/bundle.js:107884:24)
+    at performConcurrentWorkOnRoot (/static/js/bundle.js:107797:26)`;
+
+        const preparedStack = `TypeError: Cannot read properties of null (reading 'hello')
+    at Content (/static/js/bundle.js:4725:8)
+    at renderWithHooks (/static/js/bundle.js:98993:22)
+    at mountIndeterminateComponent (/static/js/bundle.js:102964:17)`;
+
+        expect(prepareErrorStack(stack, {trim: false, maxLength: 3})).toBe(preparedStack);
+    });
+
+    test('Trims data if needed', () => {
+        const stack = `TypeError: Cannot read properties of null (reading 'hello')
+    at Content (/static/js/bundle.js:4725:8)`;
+        const preparedStack = `TypeError: Cannot read properties of null (reading 'hello')
+at Content (/static/js/bundle.js:4725:8)`;
```

Comment:
I suppose when the code changes - lines and columns in bundle may change as well and the code will fail

Reply:
This code have hardcoded error string, how can it fail unless parser function is changed?